crates/core/src/host/instance_env.rs

```diff
@@ -170,6 +170,24 @@ impl InstanceEnv {
         NonZeroU32::new(count).ok_or(NodesError::ColumnValueNotFound)
     }
 
+    /// Deletes all rows in the table identified by `table_id`
+    /// where the rows match one in `relation`
+    /// which is a bsatn encoding of `Vec<ProductValue>`.
+    ///
+    /// Returns an error if no rows were deleted.
+    #[tracing::instrument(skip(self, relation))]
+    pub fn delete_by_rel(&self, table_id: TableId, relation: &[u8]) -> Result<u32, NodesError> {
```

Comment:
If we keep it, I think the optimization favoring 1 row should be documented at least here.

Reply:
I'm fine with this in doc comments for now. My thought process in the long run is that I would hope a query optimizer could be written against a documented interface. Like, I have that target audience for documentation in mind when making that original comment.